src/Microsoft.TemplateEngine.Core/Util/StreamProxy.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.IO;
+
+namespace Microsoft.TemplateEngine.Core.Util
+{
+    /// <summary>
+    /// Simple implementation of stream proxy to be used in case where destination stream is being re-read and adjusted in place multiple times.
+    /// Direct I/O usage is unnecessarily costly in those situations (even with BCL buffered streams).
+    /// This implementation flips to destination stream as soon as the cumulative size of the stream exceeds fixed threshold.
+    /// </summary>
+    internal class StreamProxy : Stream
```

Comment:
would it be beneficial to add some unit tests for this class? to see if correct stream is being used